apps/api/src/modules/auth/strategies/jwt.strategy.ts

```diff
@@ -0,0 +1,24 @@
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { JwtPayload } from 'src/common/constants/jwtInterface';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor(private readonly configService: ConfigService) {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      ignoreExpiration: false,
+      secretOrKey: configService.getOrThrow('JWT_SECRET'),
+    });
+  }
+
+  async validate(payload: JwtPayload): Promise<JwtPayload> {
+    return {
+      userId: payload.userId,
+      username: payload.username,
+      role: payload.role,
```

Comment:
Do we have roles right now?

Reply:
we have 2 role 

normal user - 0 
admin user - 1